db.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"fmt"
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+	"log"
+)
+
+type CertDetail struct {
+	gorm.Model
+	Domain string `gorm:"unique"`
+	Value  string
+}
+
+type DomainAltname struct {
+	gorm.Model
+	Domain string `gorm:"unique"`
+	Value  string
+}
+
+type UserInfo struct {
+	gorm.Model
+	Email string `gorm:"unique"`
+	Value string
+}
+
+func db(dbArgs string) (*gorm.DB, error) {
+	db, err := gorm.Open("postgres", dbArgs)
+	if err != nil {
+		fmt.Println(err)
+		panic("failed to connect to certManager database")
+	}
+
+	return db, err
+}
+
+/**
+Get Domain AltNames via domain
+*/
+func getAltNames(domain string, d *gorm.DB) (altNamesRaw []byte, err error) {
+	log.Printf("Retreving domain alt-names from database for domain (%s)", domain)
+
+	var altNames DomainAltname
+	d.Where(&DomainAltname{Domain: domain}).First(&altNames)
+	altNamesRaw = []byte(altNames.Value)
+	return altNamesRaw, err
+}
+
+/**
+Get User Info via email
+*/
+func getUserInfo(email string, d *gorm.DB) (userInfoRaw []byte, err error) {
+	log.Printf("Retreving user info email (%s) from database", email)
+
+	var userInfo UserInfo
+	d.Where(&UserInfo{Email: email}).Find(&userInfo)
+	userInfoRaw = []byte(userInfo.Value)
+	return userInfoRaw, err
+}
+
+/**
+Get Certificate Details via Domain
+*/
+func getCertDetails(domain string, d *gorm.DB) (certDetailsRaw []byte, err error) {
+	log.Printf("Retreving certificate details from database for domain (%s)", domain)
+
+	var certDetails CertDetail
+	d.Where(&CertDetail{Domain: domain}).First(&certDetails)
+	certDetailsRaw = []byte(certDetails.Value)
+	return certDetailsRaw, err
+}
+
+/**
+Save User Information key email, value userInfo
+*/
+func addUserInfo(email string, userInfo []byte, d *gorm.DB) (err error) {
+	log.Printf("Saving user info email (%s) to database", email)
```

Comment:
Spam and errors.

Reply:
Removing spam, see above comment about error handling.